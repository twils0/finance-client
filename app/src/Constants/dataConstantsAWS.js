// eslint-disable-next-line
export const actionTypes = {
  SET_AWS_STATUS: 'DATA_AUTH_AWS_STATUS',
  SET_AWS_USER: 'DATA_AUTH_SET_AWS_USER',
  RESET_AWS_STATE: 'DATA_AUTH_RESET_AWS_STATE',
};
