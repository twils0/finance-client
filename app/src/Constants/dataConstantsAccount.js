export const actionTypes = {
  SET_ACCOUNT_STATUS: 'DATA_ACCOUNT_SET_ACCOUNT_STATUS',
  SET_FIELD: 'DATA_ACCOUNT_SET_FIELD',
  SET_FIELDS: 'DATA_ACCOUNT_SET_FIELDS',
  RESET_ACCOUNT_STATE: 'DATA_ACCOUNT_RESET_ACCOUNT_STATE',
};

export const statusNames = {
  AWS_FIELDS: 'accountStatusAWSFields',
  STRIPE_FIELDS: 'accountStatusStripeFields',
  UPDATE_STRIPE_FIELDS_TOKEN: 'accountStatusUpdateStripeFieldsToken',
  UPDATE_STRIPE_FIELDS_REQUEST: 'accountStatusUpdateStripeFieldsRequest',
  UPDATE_DB_FIELDS: 'accountStatusUpdateDBFields',
  UPDATE_AWS_FIELDS: 'accountStatusUpdateAWSFields',
};

export const fieldNames = {
  NAME: 'accountName',
  EMAIL: 'accountEmail',
  EMAIL_ADDITIONAL: 'accountEmailAdditional',
  PHONE: 'accountPhone',
  NAME_ON_CARD: 'accountNameOnCard',
  PROMO_CODE: 'accountPromoCode',
  PROMO_CODE_VALID: 'accountPromoCodeValid',
  CUSTOMER: 'accountCustomer',
};
