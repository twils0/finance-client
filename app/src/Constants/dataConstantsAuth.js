export const actionTypes = {
  SET_AUTHENTICATED: 'DATA_AUTH_SET_AUTHENTICATED',
  SET_REDIRECT_URL: 'DATA_AUTH_SET_REDIRECT_URL',
  SET_AUTH_STATUS: 'DATA_AUTH_SET_AUTH_STATUS',
  SET_CODE_TYPE: 'DATA_AUTH_SET_CODE_TYPE',
  RESET_AUTH_STATE: 'DATA_AUTH_RESET_AUTH_STATE',
};

export const statusNames = {
  LOGIN: 'authStatusLogin',
  SIGN_OUT_DEVICES: 'authStatusSignOutDevices',
  LOGIN_MFA: 'authStatusLoginMFA',
  SIGN_UP: 'authStatusSignUp',
  VERIFY_PHONE: 'authStatusVerifyPhone',
  VERIFY_PHONE_CODE: 'authStatusVerifyPhoneCode',
  VERIFY_EMAIL: 'authStatusVerifyEmail',
  VERIFY_EMAIL_LINK: 'authStatusVerifyEmailLink',
  FORGOT_PASSWORD: 'authStatusForgotPassword',
  RESET_PASSWORD: 'authStatusResetPassword',
  CHANGE_PASSWORD: 'authStatusChangePassword',
  DELETE_ACCOUNT: 'authStatusDeleteAccount',
  LOGOUT: 'authStatusLogout',
};

export const codeTypeNames = {
  VERIFY_PHONE: 'verifyPhone',
  VERIFY_EMAIL: 'verifyEmail',
  VERIFY_EMAIL_ADDITIONAL: 'verifyEmailAdditional',
};
