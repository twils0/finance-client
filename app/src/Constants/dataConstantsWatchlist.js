export const actionTypes = {
  SET_WATCHLIST_STATUS: 'DATA_WATCHLIST_SECURITIES_STATUS',
  SET_SECURITIES_CURRENT: 'DATA_WATCHLIST_SET_SECURITIES_CURRENT',
  SET_SECURITIES_LIST: 'DATA_WATCHLIST_SET_SECURITIES_LIST',
  SET_SECURITIES_ALL: 'DATA_WATCHLIST_SET_SECURITIES_ALL',
  SET_SECURITY: 'DATA_WATCHLIST_SET_SECURITY',
  SET_SECURITY_DATA: 'DATA_WATCHLIST_SET_SECURITY_DATA',
  SET_SECURITY_SUBSCRIBE: 'DATA_WATCHLIST_SET_SECURITY_SUBSCRIBE',
  RESET_WATCHLIST_STATE: 'DATA_ACCOUNT_RESET_WATCHLIST_STATE',
};

export const codeTypeNames = {
  CONFIRM_PHONE: 'confirmPhone',
  CONFIRM_EMAIL: 'confirmEmail',
};

export const statusNames = {
  SECURITIES: 'watchlistStatusSecurities',
  GET_SECURITY_DATA: 'watchlistStatusGetSecurityData',
  UPDATE_SECURITIES: 'watchlistStatusUpdateSecurities',
};
