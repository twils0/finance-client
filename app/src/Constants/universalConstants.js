import stripeConfig from '../../../stripe.config.json';
import dbConfig from '../../../db.config.json';
import iexConfig from '../../../iex.config.json';

export const baseURLs = {
  STRIPE: stripeConfig.url,
  DB: dbConfig.url,
  IEX: iexConfig.urls.api,
  IEX_DEV: iexConfig.urls.dev,
  IEX_TERMS: iexConfig.urls.terms,
};

export const URLs = {
  CUSTOMERS: '/customers',
  USERS: '/users',
  EMAILS: '/emails',
  SECURITIES: '/securities',
  SECURITIES_SEARCH: '/securities/search',
  IEX_COMPANY: tickerCusip => `/stock/${tickerCusip}/company`,
  IEX_QUOTE: tickerCusip => `/stock/${tickerCusip}/quote`,
};

export const axiosConfig = {
  STRIPE: {
    baseURL: baseURLs.STRIPE,
    timeout: 30000,
  },
  DB: {
    baseURL: baseURLs.DB,
    timeout: 30000,
  },
  IEX: {
    baseURL: baseURLs.IEX,
    timeout: 30000,
  },
};

export const pathNames = {
  DEFAULT: '*',
  LOGIN: '/login',
  SIGN_UP: '/signup',
  CODE: '/code',
  CODE_VERIFY_EMAIL: '/code/:verificationId?',
  TERMS: '/terms',
  INTERNAL: '/internal',
  WATCHLIST: '/internal/watchlist',
  WATCHLIST_SECURITY_ID: '/internal/watchlist/:securityId?',
  ACCOUNT: '/internal/account',
};

export const requestStatusTypes = {
  IDLE: 'IDLE',
  LOADING: 'LOADING',
  SUCCESS: 'SUCCESS',
  ERROR: 'ERROR',
};
