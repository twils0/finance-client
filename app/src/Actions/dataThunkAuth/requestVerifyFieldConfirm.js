import { Auth } from 'aws-amplify';
import raven from 'raven-js';

import { requestStatusTypes } from '../../Constants/universalConstants';
import { statusNames, codeTypeNames } from '../../Constants/dataConstantsAuth';
import { setAuthStatus, setCodeType } from '../dataActionsAuth';
import requestLogout from './requestLogout';
import requestAWSUser from '../dataThunkAWS/requestAWSUser';

import handleErrorCatch from '../../handleErrorCatch';

import { demo } from '../../../../mode.config.json';

const requestVerifyFieldConfirm = (payload) => {
  if (!Object.prototype.hasOwnProperty.call(payload, 'field')) {
    throw new Error(`Please enter a value for the 'field' key - ${JSON.stringify(payload)}`);
  }
  if (payload.field !== 'phone_number') {
    throw new Error(
      `Please enter 'phone_number' as the value for the 'field' key - ${JSON.stringify(payload)}`,
    );
  }
  if (!Object.prototype.hasOwnProperty.call(payload, 'code')) {
    throw new Error(`Please enter a value for the 'code' key - ${JSON.stringify(payload)}`);
  }

  return async (dispatch, getState) => {
    const state = getState();

    if (
      payload.field === 'phone_number'
      && state.data.auth.status[statusNames.VERIFY_PHONE_CODE].status !== requestStatusTypes.LOADING
    ) {
      switch (payload.field) {
        case 'phone_number':
          dispatch(
            setAuthStatus({
              id: statusNames.VERIFY_PHONE_CODE,
              status: requestStatusTypes.LOADING,
            }),
          );
          break;
        default:
          break;
      }

      let { user } = state.data.aws;

      try {
        if (!user) {
          ({ user } = await dispatch(requestAWSUser()));
        }

        if (!demo) {
          await Auth.verifyUserAttributeSubmit(user, payload.field, payload.code);
        }
      } catch (errorCatch) {
        const error = handleErrorCatch(errorCatch);

        switch (payload.field) {
          case 'phone_number':
            dispatch(
              setAuthStatus({
                id: statusNames.VERIFY_PHONE_CODE,
                status: requestStatusTypes.ERROR,
              }),
            );
            break;
          default:
            break;
        }

        if (typeof error === 'object' && error && error.code === 'NotAuthorizedException') {
          dispatch(requestLogout());

          return null;
        }

        if (
          typeof error === 'object'
          && error
          && (error.code === 'LimitExceededException' || error.code === 'CodeMismatchException')
        ) {
          return Promise.reject({ field: payload.field, error });
        }

        if (!demo) {
          raven.captureException(error, {
            logger: 'requestVerifyFieldConfirm',
          });
        }

        return Promise.reject({ field: payload.field, error });
      }

      switch (payload.field) {
        case 'phone_number':
          dispatch(setCodeType({ id: codeTypeNames.VERIFY_PHONE, needed: false }));
          dispatch(
            setAuthStatus({
              id: statusNames.VERIFY_PHONE_CODE,
              status: requestStatusTypes.SUCCESS,
            }),
          );
          break;
        default:
          break;
      }
    }

    return null;
  };
};

export default requestVerifyFieldConfirm;
