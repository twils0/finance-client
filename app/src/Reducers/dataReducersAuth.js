import { combineReducers } from 'redux';

import { requestStatusTypes, pathNames } from '../Constants/universalConstants';
import { actionTypes, statusNames, codeTypeNames } from '../Constants/dataConstantsAuth';

export const initialState = {
  authenticated: false,
  redirectURL: pathNames.WATCHLIST,
  status: {
    [statusNames.LOGIN]: {
      id: statusNames.LOGIN,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.SIGN_OUT_DEVICES]: {
      id: statusNames.SIGN_OUT_DEVICES,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.LOGIN_MFA]: {
      id: statusNames.LOGIN_MFA,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.SIGN_UP]: {
      id: statusNames.SIGN_UP,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.VERIFY_PHONE]: {
      id: statusNames.VERIFY_PHONE,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.VERIFY_PHONE_CODE]: {
      id: statusNames.VERIFY_PHONE_CODE,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.VERIFY_EMAIL]: {
      id: statusNames.VERIFY_EMAIL,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.VERIFY_EMAIL_LINK]: {
      id: statusNames.VERIFY_EMAIL_LINK,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.FORGOT_PASSWORD]: {
      id: statusNames.FORGOT_PASSWORD,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.RESET_PASSWORD]: {
      id: statusNames.RESET_PASSWORD,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.CHANGE_PASSWORD]: {
      id: statusNames.CHANGE_PASSWORD,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.LOGOUT]: {
      id: statusNames.LOGOUT,
      status: requestStatusTypes.SUCCESS,
    },
    [statusNames.DELETE_ACCOUNT]: {
      id: statusNames.DELETE_ACCOUNT,
      status: requestStatusTypes.SUCCESS,
    },
  },
  codeTypes: {
    [codeTypeNames.VERIFY_PHONE]: {
      id: codeTypeNames.VERIFY_PHONE,
      needed: true,
    },
    [codeTypeNames.VERIFY_EMAIL]: {
      id: codeTypeNames.VERIFY_EMAIL,
      needed: true,
    },
    [codeTypeNames.VERIFY_EMAIL_ADDITIONAL]: {
      id: codeTypeNames.VERIFY_EMAIL_ADDITIONAL,
      needed: true,
    },
  },
};

export const authenticated = (state = initialState.authenticated, action) => {
  switch (action.type) {
    case actionTypes.SET_AUTHENTICATED:
      return action.payload.authenticated;
    case actionTypes.RESET_AUTH_STATE:
      return initialState.authenticated;
    default:
      return state;
  }
};

export const redirectURL = (state = initialState.redirectURL, action) => {
  switch (action.type) {
    case actionTypes.SET_REDIRECT_URL:
      return action.payload.redirectURL;
    default:
      return state;
  }
};

export const status = (state = initialState.status, action) => {
  switch (action.type) {
    case actionTypes.SET_AUTH_STATUS:
      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          ...action.payload,
        },
      };
    case actionTypes.RESET_AUTH_STATE:
      return initialState.status;
    default:
      return state;
  }
};

export const codeTypes = (state = initialState.codeTypes, action) => {
  switch (action.type) {
    case actionTypes.SET_CODE_TYPE:
      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          ...action.payload,
        },
      };
    case actionTypes.RESET_AUTH_STATE:
      return initialState.codeTypes;
    default:
      return state;
  }
};

const dataReducersAuth = combineReducers({
  authenticated,
  redirectURL,
  status,
  codeTypes,
});

export default dataReducersAuth;
