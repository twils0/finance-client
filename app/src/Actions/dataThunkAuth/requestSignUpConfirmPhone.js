import { Auth } from 'aws-amplify';
import raven from 'raven-js';

import handleErrorCatch from '../../handleErrorCatch';
import { requestStatusTypes } from '../../Constants/universalConstants';
import { statusNames, codeTypeNames } from '../../Constants/dataConstantsAuth';
import { setCodeType, setAuthStatus } from '../dataActionsAuth';
import requestVerifyFieldConfirm from './requestVerifyFieldConfirm';

import { demo } from '../../../../mode.config.json';

const requestSignUpConfirmPhone = (payload) => {
  if (!Object.prototype.hasOwnProperty.call(payload, 'email')) {
    throw Error(`Please enter a value for the 'email' key - ${payload}`);
  }
  if (!Object.prototype.hasOwnProperty.call(payload, 'code')) {
    throw new Error(`Please enter a value for the 'code' key - ${JSON.stringify(payload)}`);
  }

  return async (dispatch, getState) => {
    const state = getState();

    if (
      state.data.auth.status[statusNames.VERIFY_PHONE_CODE].status !== requestStatusTypes.LOADING
    ) {
      dispatch(
        setAuthStatus({
          id: statusNames.VERIFY_PHONE_CODE,
          status: requestStatusTypes.LOADING,
        }),
      );

      try {
        if (!demo) {
          await Auth.confirmSignUp(payload.email, payload.code);
        }
      } catch (errorCatch) {
        const error = handleErrorCatch(errorCatch);

        if (error.code === 'NotAuthorizedException' && error.message.indexOf('UNCONFIRMED') > -1) {
          try {
            await dispatch(
              requestVerifyFieldConfirm({
                field: 'phone_number',
                code: payload.code,
              }),
            );

            dispatch(setCodeType({ id: codeTypeNames.VERIFY_PHONE_CODE, needed: false }));
            dispatch(
              setAuthStatus({
                id: statusNames.VERIFY_PHONE_CODE,
                status: requestStatusTypes.SUCCESS,
              }),
            );

            return null;
          } catch (errorCatch2) {
            const error2 = handleErrorCatch(errorCatch2);

            dispatch(
              setAuthStatus({
                id: statusNames.VERIFY_PHONE_CODE,
                status: requestStatusTypes.ERROR,
              }),
            );

            return Promise.reject(error2.error);
          }
        }

        dispatch(
          setAuthStatus({
            id: statusNames.VERIFY_PHONE_CODE,
            status: requestStatusTypes.ERROR,
          }),
        );

        if (
          typeof error === 'object'
          && error
          && (error.code === 'LimitExceededException'
            || error.code === 'ExpiredCodeException'
            || error.code === 'CodeMismatchException')
        ) {
          return Promise.reject(error);
        }

        if (!demo) {
          raven.captureException(error, {
            logger: 'requestSignUpConfirmPhone',
          });
        }

        return Promise.reject(error);
      }

      dispatch(setCodeType({ id: codeTypeNames.VERIFY_PHONE_CODE, needed: false }));
      dispatch(
        setAuthStatus({
          id: statusNames.VERIFY_PHONE_CODE,
          status: requestStatusTypes.SUCCESS,
        }),
      );
    }

    return null;
  };
};

export default requestSignUpConfirmPhone;
