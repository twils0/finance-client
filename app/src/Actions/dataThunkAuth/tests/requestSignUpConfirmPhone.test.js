import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Auth } from 'aws-amplify';

import { initialState as initialStateAuth } from '../../../Reducers/dataReducersAuth';
import { requestStatusTypes } from '../../../Constants/universalConstants';
import {
  actionTypes as actionTypesAuth,
  statusNames,
  codeTypeNames,
} from '../../../Constants/dataConstantsAuth';
import requestSignUpConfirmPhone from '../requestSignUpConfirmPhone';
import requestVerifyFieldConfirm from '../requestVerifyFieldConfirm';

jest.mock('aws-amplify', () => ({
  Auth: {
    confirmSignUp: jest.fn(),
  },
}));
jest.mock('../requestVerifyFieldConfirm', () => jest.fn());

const middleware = [thunk];
const mockStore = configureMockStore(middleware);

const signUpConfirmPayload = {
  email: 'testEmail',
  code: '123456',
};

describe('dataThunkAuth', () => {
  describe('requestSignUpConfirmPhone', () => {
    afterEach(() => {
      Auth.confirmSignUp.mockReset();
      requestVerifyFieldConfirm.mockReset();
    });

    it("fails and throws an error when missing 'email' key in payload", async () => {
      const store = mockStore({ data: { auth: initialStateAuth } });

      const emptyPayload = {};

      try {
        await store.dispatch(requestSignUpConfirmPhone(emptyPayload));
      } catch (errorCatch) {
        expect(errorCatch).toMatchSnapshot();
      }

      expect(Auth.confirmSignUp).not.toBeCalled();
      expect(requestVerifyFieldConfirm).not.toBeCalled();
    });

    it("fails when provided a payload with an 'code' key that is not an object", async () => {
      const store = mockStore({ data: { auth: initialStateAuth } });

      Auth.confirmSignUp.mockReturnValue(Promise.resolve());

      const wrongPayload = { email: 'testEmail' };

      try {
        await store.dispatch(requestSignUpConfirmPhone(wrongPayload));
      } catch (errorCatch) {
        expect(errorCatch).toMatchSnapshot();
      }

      expect(Auth.confirmSignUp).not.toBeCalled();
      expect(requestVerifyFieldConfirm).not.toBeCalled();
    });

    it("fails and requestVerifyFieldConfirm throws an error, when in turn axios throws 'NotAuthorizedException' error, containing 'UNCONFIRMED' in the message", async () => {
      const store = mockStore({ data: { auth: initialStateAuth } });

      const error = { code: 'NotAuthorizedException', message: 'testMessage UNCONFIRMED' };
      const error2 = {
        field: 'phone_number',
        error: { code: 'testError', message: 'testMessage' },
      };
      const verifyPayload = {
        field: 'phone_number',
        code: signUpConfirmPayload.code,
      };
      const expectedActions = [
        {
          type: actionTypesAuth.SET_AUTH_STATUS,
          payload: {
            id: statusNames.VERIFY_PHONE_CODE,
            status: requestStatusTypes.LOADING,
          },
        },
        {
          type: actionTypesAuth.SET_AUTH_STATUS,
          payload: {
            id: statusNames.VERIFY_PHONE_CODE,
            status: requestStatusTypes.ERROR,
          },
        },
      ];

      Auth.confirmSignUp.mockReturnValue(Promise.reject(error));
      requestVerifyFieldConfirm.mockReturnValue(() => Promise.reject(error2));

      try {
        await store.dispatch(requestSignUpConfirmPhone(signUpConfirmPayload));
      } catch (errorCatch) {
        expect(errorCatch).toEqual(error2.error);
      }

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
      expect(Auth.confirmSignUp).toBeCalledWith(
        signUpConfirmPayload.email,
        signUpConfirmPayload.code,
      );
      expect(requestVerifyFieldConfirm).toBeCalledWith(verifyPayload);
    });

    it("calls requestVerifyFieldConfirm when axios throws 'NotAuthorizedException' error, containing 'UNCONFIRMED' in the message", async () => {
      const store = mockStore({ data: { auth: initialStateAuth } });

      const error = { code: 'NotAuthorizedException', message: 'testMessage UNCONFIRMED' };
      const verifyPayload = {
        field: 'phone_number',
        code: signUpConfirmPayload.code,
      };
      const expectedActions = [
        {
          type: actionTypesAuth.SET_AUTH_STATUS,
          payload: {
            id: statusNames.VERIFY_PHONE_CODE,
            status: requestStatusTypes.LOADING,
          },
        },
        {
          type: actionTypesAuth.SET_CODE_TYPE,
          payload: {
            id: codeTypeNames.VERIFY_PHONE_CODE,
            needed: false,
          },
        },
        {
          type: actionTypesAuth.SET_AUTH_STATUS,
          payload: {
            id: statusNames.VERIFY_PHONE_CODE,
            status: requestStatusTypes.SUCCESS,
          },
        },
      ];

      Auth.confirmSignUp.mockReturnValue(Promise.reject(error));
      requestVerifyFieldConfirm.mockReturnValue(() => Promise.resolve());

      const result = await store.dispatch(requestSignUpConfirmPhone(signUpConfirmPayload));

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
      expect(result).toEqual(null);
      expect(Auth.confirmSignUp).toBeCalledWith(
        signUpConfirmPayload.email,
        signUpConfirmPayload.code,
      );
      expect(requestVerifyFieldConfirm).toBeCalledWith(verifyPayload);
    });

    it('fails and returns promise reject when Auth.confirmSignUp throws an error', async () => {
      const store = mockStore({ data: { auth: initialStateAuth } });

      const error = { code: 'testError', message: 'testMessage' };
      const expectedActions = [
        {
          type: actionTypesAuth.SET_AUTH_STATUS,
          payload: {
            id: statusNames.VERIFY_PHONE_CODE,
            status: requestStatusTypes.LOADING,
          },
        },
        {
          type: actionTypesAuth.SET_AUTH_STATUS,
          payload: {
            id: statusNames.VERIFY_PHONE_CODE,
            status: requestStatusTypes.ERROR,
          },
        },
      ];

      Auth.confirmSignUp.mockReturnValue(Promise.reject(error));

      try {
        await store.dispatch(requestSignUpConfirmPhone(signUpConfirmPayload));
      } catch (errorCatch) {
        expect(errorCatch).toEqual(error);
      }

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
      expect(Auth.confirmSignUp).toBeCalledWith(
        signUpConfirmPayload.email,
        signUpConfirmPayload.code,
      );
      expect(requestVerifyFieldConfirm).not.toBeCalled();
    });

    it('creates the correct actions with the correct payloads', async () => {
      const store = mockStore({ data: { auth: initialStateAuth } });

      const expectedActions = [
        {
          type: actionTypesAuth.SET_AUTH_STATUS,
          payload: {
            id: statusNames.VERIFY_PHONE_CODE,
            status: requestStatusTypes.LOADING,
          },
        },
        {
          type: actionTypesAuth.SET_CODE_TYPE,
          payload: {
            id: codeTypeNames.VERIFY_PHONE_CODE,
            needed: false,
          },
        },
        {
          type: actionTypesAuth.SET_AUTH_STATUS,
          payload: {
            id: statusNames.VERIFY_PHONE_CODE,
            status: requestStatusTypes.SUCCESS,
          },
        },
      ];

      Auth.confirmSignUp.mockReturnValue(Promise.resolve());

      const result = await store.dispatch(requestSignUpConfirmPhone(signUpConfirmPayload));

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
      expect(result).toEqual(null);
      expect(Auth.confirmSignUp).toBeCalledWith(
        signUpConfirmPayload.email,
        signUpConfirmPayload.code,
      );
      expect(requestVerifyFieldConfirm).not.toBeCalled();
    });
  });
});
