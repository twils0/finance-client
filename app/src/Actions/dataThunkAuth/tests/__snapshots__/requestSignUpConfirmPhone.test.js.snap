// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dataThunkAuth requestSignUpConfirmPhone fails and throws an error when missing 'email' key in payload 1`] = `[Error: Please enter a value for the 'email' key - [object Object]]`;

exports[`dataThunkAuth requestSignUpConfirmPhone fails when provided a payload with an 'code' key that is not an object 1`] = `[Error: Please enter a value for the 'code' key - {"email":"testEmail"}]`;
